framework/include/base/ThreadedElementLoopBase.h

```diff
@@ -159,7 +162,10 @@ ThreadedElementLoopBase<RangeType>::operator()(const RangeType & range, bool byp
       _subdomain = cur_subdomain;
 
       if (_subdomain != _old_subdomain)
+      {
+        setCurrentElem(_tid, elem);
```

Comment:
As far as fixing the value of _current_elem when we change subdomains, this is fine.